source/Serial/DfRobotBleSerial.cpp

```diff
@@ -180,6 +180,26 @@ DfRobotBleSerial::end(
     _device_collection = nullptr;
 }
 
+void
+DfRobotBleSerial::flush(
+    void
+    )
+{
+    create_task(_gatt_characteristic->WriteValueAsync(_tx->DetachBuffer(), GattWriteOption::WriteWithResponse))
+        .then([this](GattCommunicationStatus status_)
+    {
+        switch (status_) {
+        case GattCommunicationStatus::Success:
+            break;
```

Comment:
This api (as far as I can tell) gives no way to know when the write "completed". Specifically, I can't know for sure when I'm no longer in danger of getting a call to "ConnectionLost()".

Therefore, if I want to call the API twice, I have no way to know which one failed and which one passed. Usually you would just sequence them: call the first, wait until it succeeds or fails, then try the second.